src/glang/codegen/translation_unit.py

```diff
@@ -151,23 +195,29 @@ def lookup_call_expression(
         fn_name: str,
         fn_specialization: list[SpecializationItem],
         fn_args: list[TypedExpression],
+        meta: Meta,
     ) -> TypedExpression:
         if fn_name in self._builtin_callables:
-            return self._builtin_callables[fn_name](fn_specialization, fn_args)
+            return self._builtin_callables[fn_name](fn_specialization, fn_args, meta)
 
         signature, _ = self.symbol_table.lookup_function(
             fn_name, fn_specialization, fn_args, True
         )
-        return FunctionCallExpression(signature, fn_args)
+        return FunctionCallExpression(signature, fn_args, meta)
 
-    def add_static_string(self, string: str) -> StaticVariable:
+    def add_static_string(self, string: str, meta: Meta) -> StaticVariable:
         if string in self._string_cache:
             return self._string_cache[string]
 
         encoded_str, encoded_length = encode_string(string)
         str_type = AnonymousType(CharArrayDefinition(encoded_str, encoded_length))
         static_storage = StaticVariable(
-            f'string literal: "{string}"', str_type, False, encoded_str
+            f'string literal: "{string}"',
+            str_type,
+            False,
+            encoded_str,
+            meta,
+            self.di_files[0],  # FIXME this is wrong.
```

Comment:
Seems like `Meta` should have a file path

Reply:
That would be a really nice refactor